pallets/asset-registry/src/lib.rs

```diff
@@ -74,27 +84,30 @@ pub mod pallet {
 			+ MaxEncodedLen
 			+ TypeInfo;
 
-		/// Balance type
-		type Balance: Parameter
-			+ Member
-			+ AtLeast32BitUnsigned
-			+ Default
-			+ Copy
-			+ MaybeSerializeDeserialize
-			+ MaxEncodedLen;
-
 		/// Asset location type
 		type AssetNativeLocation: Parameter + Member + Default + MaxEncodedLen;
 
-		/// The maximum length of a name or symbol stored on-chain.
-		type StringLimit: Get<u32>;
+		/// Multi currency mechanism
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = Self::AssetId, Balance = Balance>;
```

Comment:
i wonder if we should just try to use fungibles traits from now one where multi currecny support is required,.

this way, we just rely on substrate instead of orml.